### PR TITLE
added corner cases for comments.go in astbuilder package

### PR DESCRIPTION
**Pull request by khareyash05**
**What this PR does / why we need it**:
Increase test coverage of `astbuillder/comments.go` to add corner cases of empty comments and text width =0 

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains tests

### CODE REVIEW

File: v2/tools/generator/internal/astbuilder/comments_test.go [hunk 1]
```diff
@@ -48,6 +49,9 @@ func TestDocumentationCommentFormatting(t *testing.T) {
 	}
 
 	for _, c := range cases {
+		if c.comment == "" {
+			continue
+		}
```
Comment (theunrepentantgeek) [inline v2/tools/generator/internal/astbuilder/comments_test.go:54]: All this does is to suppress the new test case - it's never run at all, so it adds nothing to the coverage.

File: v2/tools/generator/internal/astbuilder/comments_test.go [hunk 2]
```diff
@@ -70,9 +74,13 @@ func TestWordWrap(t *testing.T) {
 		{"this is a simple line of text", 16, []string{"this is a simple ", "line of text"}},
 		{"this is a simple line of text", 20, []string{"this is a simple ", "line of text"}},
 		{"this is a simple line of text", 21, []string{"this is a simple line ", "of text"}},
+		{"", 0, []string{}},
 	}
 
 	for _, c := range cases {
+		if c.width == 0 {
+			continue
+		}
```
Comment (theunrepentantgeek) [inline v2/tools/generator/internal/astbuilder/comments_test.go:83]: Ditto here; all you're doing is skipping the test if the passed width is zero.

**Review by theunrepentantgeek — CHANGES_REQUESTED** [review]

File: v2/tools/generator/internal/astbuilder/comments_test.go (hunk 1 above)
Comment (khareyash05) [reply v2/tools/generator/internal/astbuilder/comments_test.go:54]: Will update the same

File: v2/tools/generator/internal/astbuilder/comments_test.go [hunk 3]
```diff
@@ -70,6 +71,8 @@ func TestWordWrap(t *testing.T) {
 		{"this is a simple line of text", 16, []string{"this is a simple ", "line of text"}},
 		{"this is a simple line of text", 20, []string{"this is a simple ", "line of text"}},
 		{"this is a simple line of text", 21, []string{"this is a simple line ", "of text"}},
+		{"", 0, []string{}},
+		{"this is a sample text", 0, []string{}},
```
Comment (theunrepentantgeek) [inline v2/tools/generator/internal/astbuilder/comments_test.go:75]: We should never throw away content like this - if someone inadvertently passes a width of `0` we want to return the content anyway. For consistency with existing behaviour when width is `1`, this should be:

```suggestion
		{"this is a sample text", 0, []string{"this ", "is ", "a ", "sample ", "text"}},
```

**Review by theunrepentantgeek — CHANGES_REQUESTED** [review]
Adding coverage of edge cases is valuable, but don't lose sight of the context the methods are used within.

**Review by theunrepentantgeek — APPROVED** [review]

**Comment by theunrepentantgeek** [comment]
Thanks @khareyash05.
